src/Build/Evaluation/Evaluator.cs

```diff
@@ -1256,7 +1274,7 @@ private ICollection<P> AddEnvironmentProperties()
 
             foreach (ProjectPropertyInstance environmentProperty in _environmentProperties)
             {
-                P property = _data.SetProperty(environmentProperty.Name, ((IProperty)environmentProperty).EvaluatedValueEscaped, false /* NOT global property */, false /* may NOT be a reserved name */);
+                P property = _data.SetProperty(environmentProperty.Name, ((IProperty)environmentProperty).EvaluatedValueEscaped, false /* NOT global property */, false /* may NOT be a reserved name */, true /* IS an environment variable.*/);
```

Comment:
nit: can use argument names instead of comments.

Reply:
Changed with latest update.